common/common-rest/src/main/java/org/apache/servicecomb/common/rest/codec/RestCodec.java

```diff
@@ -61,12 +64,19 @@ public static void argsToRest(Object[] args, RestOperationMeta restOperation,
       try {
         paramValues[idx] = param.getParamProcessor().getValue(request);
       } catch (Exception e) {
-        // Avoid information leak of user input.
-        throw new InvocationException(Status.BAD_REQUEST,
-            String.format("Parameter is not valid for operation [%s]. Parameter is [%s]. Processor is [%s].",
-                restOperation.getOperationMeta().getMicroserviceQualifiedName(),
-                param.getParamName(),
-                param.getParamProcessor().getProcessorType()));
+        // Avoid information leak of user input, and add option for debug use.
+        String message = String.format("Parameter is not valid for operation [%s]. Parameter is [%s]. Processor is [%s].",
+            restOperation.getOperationMeta().getMicroserviceQualifiedName(),
+            param.getParamName(),
+            param.getParamProcessor().getProcessorType());
+        if (DynamicPropertyFactory.getInstance().getBooleanProperty(
+            RestConst.PRINT_CODEC_ERROR_MESSGAGE, false).get()) {
+          LOG.error("", e);
+          throw new InvocationException(Status.BAD_REQUEST.getStatusCode(), "", message, e);
+        } else {
+          LOG.error(message + " Add {}=true to print the details." + RestConst.PRINT_CODEC_ERROR_MESSGAGE);
```

Comment:
Here should use comma instead of plus. And since `RestConst.PRINT_CODEC_ERROR_MESSGAGE` is a const, why not combine the String by `+` directly?